### PR TITLE
[addons][screensaver] code improvement within CGUIWindowScreensaver

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

There becomes the call amount of "CServiceBroker::GetWinSystem()->GetGfxContext()" reduced, with store of it as value and to reuse on following places.

This reduce within render flow the work about calls and on "CServiceBroker::GetWinSystem()->GetGfxContext()" to have an shorter line :smirk:.

Further done a whole clang cleanup on this file.

Also done this changes within my sandbox project and as unrelated to it, comes separate request here.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/windows/GUIWindowScreensaver.cpp [hunk 1]
```diff
@@ -28,16 +28,20 @@ void CGUIWindowScreensaver::Process(unsigned int currentTime, CDirtyRegionList &
 {
   MarkDirtyRegion();
   CGUIWindow::Process(currentTime, regions);
-  m_renderRegion.SetRect(0, 0, (float)CServiceBroker::GetWinSystem()->GetGfxContext().GetWidth(), (float)CServiceBroker::GetWinSystem()->GetGfxContext().GetHeight());
+  auto& context = CServiceBroker::GetWinSystem()->GetGfxContext();
```
Comment (howie-f) [inline xbmc/windows/GUIWindowScreensaver.cpp:30]: `const auto&` :) please

**Review by howie-f — APPROVED** [review]
thanks, looks good to me

File: xbmc/windows/GUIWindowScreensaver.cpp (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/windows/GUIWindowScreensaver.cpp]: Have changed this line, as `GetWidth()` and `GetHeight()` defined as `const` it works. On the other two places not possible, as `ApplyStateBlock` and `CaptureStateBlock` do something.

File: xbmc/windows/GUIWindowScreensaver.cpp [hunk 2]
```diff
@@ -53,6 +57,8 @@ EVENT_RESULT CGUIWindowScreensaver::OnMouseEvent(const CPoint &point, const CMou
 
 bool CGUIWindowScreensaver::OnMessage(CGUIMessage& message)
 {
+  auto& context = CServiceBroker::GetWinSystem()->GetGfxContext();
```
Comment (ksooo) [inline xbmc/windows/GUIWindowScreensaver.cpp:59]: Not sure I would unconditionally make this call here as you actually need the value only in two out of many `case`s... I would make the call only when actually needed, from within the `case`.

Comment (garbear) [reply xbmc/windows/GUIWindowScreensaver.cpp:59]: For a regular variable I'd totally agree. However for a global I like to "break" least privilege as little as possible and as visibly as possible. See how I put the scope breakage at the beginning of the functions in [ProfileManager.cpp](https://github.com/xbmc/xbmc/blob/master/xbmc/profiles/ProfileManager.cpp#L377). Scope breakage happens all in one stanza there before everything else.

**Review by garbear — APPROVED** [review]
No issues that I can see, other than what @ksooo addressed.

**Pull request merged**